server/api/crud/projects.py

```diff
@@ -384,3 +388,46 @@ async def _calculate_pipelines_counters(
             if pipeline["status"] not in mlrun.run.RunStatuses.stable_statuses():
                 project_to_running_pipelines_count[pipeline["project"]] += 1
         return project_to_running_pipelines_count
+
+    @staticmethod
+    def _wait_for_nuclio_project_deletion(
+        project_name: str,
+        session: sqlalchemy.orm.Session,
+        auth_info: mlrun.common.schemas.AuthInfo = mlrun.common.schemas.AuthInfo(),
+    ):
+        if not mlrun.config.config.nuclio_dashboard_url:
+            return
+
+        nuclio_client = server.api.utils.clients.nuclio.Client(auth_info)
+
+        def _check_nuclio_project_deletion():
+            try:
+                nuclio_client.get_project(session, project_name)
+                raise Exception(
+                    f"Project not deleted in nuclio yet. Project: {project_name}"
+                )
+            except mlrun.errors.MLRunNotFoundError:
+                logger.debug(
+                    "Nuclio project deleted",
+                    project_name=project_name,
+                )
+                return True
```

Comment:
```suggestion
            try:
                nuclio_client.get_project(session, project_name)
            except mlrun.errors.MLRunNotFoundError:
                logger.debug(
                    "Nuclio project deleted",
                    project_name=project_name,
                )
                return True
            else:
                raise Exception(
                    f"Project not deleted in nuclio yet. Project: {project_name}"
                )
```
Better to raise outside of the try